python/src/phylanx.cpp

```diff
@@ -497,36 +535,390 @@ PYBIND11_MODULE(_phylanx, m)
+            return hpx::threads::run_as_hpx_thread([&]() {
+                using namespace phylanx::execution_tree;
+                return primitive{hpx::local_new<primitives::variable>(
+                    phylanx::ir::node_data<double>{m})};
+            });
+        },
+        "create a new variable from a matrix of zeros");
+
+    execution_tree.def(
+        "phylisp_eval", expression_compiler<>, expression_compiler_help);
+    execution_tree.def(
+        "phylisp_eval", expression_compiler<double>, expression_compiler_help);
+    execution_tree.def("phylisp_eval",
+        expression_compiler<phylanx::execution_tree::primitive const&>,
+        expression_compiler_help);
+    execution_tree.def("phylisp_eval",
+        expression_compiler<phylanx::execution_tree::primitive const&,
+            phylanx::execution_tree::primitive const&>,
+        expression_compiler_help);
+    execution_tree.def("phylisp_eval",
+        expression_compiler<phylanx::execution_tree::primitive const&, double>,
+        expression_compiler_help);
+    execution_tree.def("phylisp_eval",
+        expression_compiler<double, phylanx::execution_tree::primitive const&>,
+        expression_compiler_help);
+    execution_tree.def("phylisp_eval",
+        expression_compiler<double, double>,
+        expression_compiler_help);
+    execution_tree.def("phylisp_eval",
+        expression_compiler<phylanx::execution_tree::primitive const&,
+            phylanx::execution_tree::primitive const&,
+            phylanx::execution_tree::primitive const&>,
+        expression_compiler_help);
+    execution_tree.def("phylisp_eval",
+        expression_compiler<double, phylanx::execution_tree::primitive const&,
+            phylanx::execution_tree::primitive const&>,
+        expression_compiler_help);
+    execution_tree.def("phylisp_eval",
+        expression_compiler<phylanx::execution_tree::primitive const&, double,
+            phylanx::execution_tree::primitive const&>,
+        expression_compiler_help);
+    execution_tree.def("phylisp_eval",
+        expression_compiler<phylanx::execution_tree::primitive const&,
+            phylanx::execution_tree::primitive const&, double>,
+        expression_compiler_help);
+    execution_tree.def("phylisp_eval",
+        expression_compiler<double, double,
+            phylanx::execution_tree::primitive const&>,
+        expression_compiler_help);
+    execution_tree.def("phylisp_eval",
+        expression_compiler<double, phylanx::execution_tree::primitive const&,
+            double>,
+        expression_compiler_help);
+    execution_tree.def("phylisp_eval",
+        expression_compiler<phylanx::execution_tree::primitive const&, double,
+            double>,
+        expression_compiler_help);
+    execution_tree.def("phylisp_eval",
+        expression_compiler<double, double, double>,
+        expression_compiler_help);
+
+    execution_tree.def("add",
+        [](phylanx::execution_tree::primitive const& obj1,
```

Comment:
Same here, using `primitive_argument_type` as the parameter types might be better.